### PR TITLE
delvars!, delconstrs! and chgcoeffs

### PR DESCRIPTION
**Pull request by joaquimg**
With tests

### CODE REVIEW

**Comment by mlubin** [comment]
Docs?

File: doc/lpqcqp.rst [hunk 1]
```diff
@@ -93,6 +93,9 @@ to indicate equality constraints.
     objective coefficient ``objcoef``. This is equivalent to calling the
     above method with empty arrays for the constraint coefficients.
 
+.. function:: delvars!(m::AbstractLinearQuadraticModel, idxs)
+
+    Removes the variables with indexes in the vector idxs.
```
Comment (mlubin) [inline doc/lpqcqp.rst]: What happens to the remaining variables in the model? Should be explicit that they are renumbered.

**Comment by mlubin** [comment]
Opening for bikeshedding on the names (@chriscoey)

**Comment by chriscoey** [comment]
why me? i don't even own a two-wheeled transportation device, let alone care about the hue of its nonexistent storage facility

**Comment by joehuchette** [comment]
``changecoefficients`` or ``changecoeffs``? ``chg`` makes me cringe

**Comment by joehuchette** [comment]
``deletevars!`` and ``deleteconstrs``?

**Comment by joaquimg** [comment]
I like `changecoeffs`, indeed `chg` is too much abbreviation.

However, I would vote for `del` instead of `delete`, since: its self reasonably self contained; `vars` and `constrs` are already abbreviated; and is analog to `add` (although `add` is the full word)...

File: doc/solverinterface.rst [hunk 2]
```diff
@@ -107,6 +107,10 @@ All abstract model types inherit from the abstract type ``AbstractMathProgModel`
     with licensing restrictions on the number of concurrent runs.
     Users must discard the model object after this method is invoked.
 
+.. function:: copy(m::AbstractMathProgModel)
```
Comment (joehuchette) [inline doc/solverinterface.rst]: Is this new as well?

Comment (joaquimg) [inline doc/solverinterface.rst]: Yep, that is new, all the solvers had the copy function for their on Model types, but there was nothing for `AbstractMathProgModel` in MathProgBase. There was even some discussion about what to do with callbacks, I decided to no copying them and warning the user. See JuliaOpt/Gurobi.jl#78

Comment (joehuchette) [inline doc/solverinterface.rst]: I'm fine with adding it, but it should be exported and tested

Comment (joaquimg) [inline doc/solverinterface.rst]: I added something at https://github.com/joaquimg/MathProgBase.jl/blob/master/test/linproginterface.jl what do you think?

File: doc/lpqcqp.rst [hunk 3]
```diff
@@ -93,6 +93,13 @@ to indicate equality constraints.
     objective coefficient ``objcoef``. This is equivalent to calling the
     above method with empty arrays for the constraint coefficients.
 
+.. function:: delvars!(m::AbstractLinearQuadraticModel, idxs)
+
+    Removes the variables with indexes in the vector idxs. The remaing variables in the model
```
Comment (joehuchette) [inline doc/lpqcqp.rst]: *remaining

File: doc/lpqcqp.rst [hunk 4]
```diff
@@ -102,6 +109,20 @@ to indicate equality constraints.
     coefficients, and the ``varidx`` vector contains the indices of the corresponding
     variables.
 
+.. function:: delconstrs!(m::AbstractLinearQuadraticModel, idxs)
+
+    Removes the constraints with indexes in the vector idxs.The remaing constraints in the model
```
Comment (joehuchette) [inline doc/lpqcqp.rst]: *remaining

File: doc/solverinterface.rst (hunk 2 above)
Comment (joehuchette) [inline doc/solverinterface.rst]: looks fine to me

File: doc/lpqcqp.rst [hunk 5]
```diff
@@ -93,6 +93,13 @@ to indicate equality constraints.
     objective coefficient ``objcoef``. This is equivalent to calling the
     above method with empty arrays for the constraint coefficients.
 
+.. function:: delvars!(m::AbstractLinearQuadraticModel, idxs)
+
+    Removes the variables with indexes in the vector idxs. The remaining variables in the model
+    are renumbered. Let, for example, before deletion there be six
+    variables p, q, r, s, t, u with ordinal numbers 1, 2, 3, 4, 5, 6, and let variables
+    p, q, s have been deleted. Then after deletion the remaining variables r, t, u
+    are assigned new ordinal numbers 1, 2, 3.
```
Comment (mlubin) [inline doc/lpqcqp.rst]: The wording here is confusing. There's no concept of symbolic names for variables in MPB. I would say something like, "suppose we delete indices 1, 2, and 4, then there will be 3 variables remaining in the model, those which were previously numbered 3, 5, and 6."

**Comment by joaquimg** [comment]
What about naming?

I prefer `changecoeffs!`, `delvars!` and `delconstrs!`.

Do you want to go for `deletevars!` and `deleteconstrs!`?

**Comment by mlubin** [comment]
> I prefer ``changecoeffs!``, ``delvars!`` and ``delconstrs!``.

Ok with me

File: doc/lpqcqp.rst [hunk 6]
```diff
@@ -102,6 +109,20 @@ to indicate equality constraints.
     coefficients, and the ``varidx`` vector contains the indices of the corresponding
     variables.
 
+.. function:: delconstrs!(m::AbstractLinearQuadraticModel, idxs)
+
+    Removes the constraints with indexes in the vector idxs.The remaining constraints in the model
```
Comment (mlubin) [inline doc/lpqcqp.rst]: "indices" to be consistent. Missing space at "idxs.The"

**Comment by joaquimg** [comment]
Also changed in: joaquimg/Gurobi.jl@22d888701e6cd7ff7f687de5470511d2ff934725 and JuliaOpt/Xpress.jl@1bf99ebb9b73d975d1c14e0991a7afdfce60dbca

**Comment by joehuchette** [comment]
Works with me as well.






On Tue, Jan 3, 2017 at 2:35 PM -0300, "Miles Lubin" <notifications@github.com> wrote:













I prefer changecoeffs!, delvars! and delconstrs!.



Ok with me

File: doc/lpqcqp.rst [hunk 7]
```diff
@@ -93,6 +93,13 @@ to indicate equality constraints.
     objective coefficient ``objcoef``. This is equivalent to calling the
     above method with empty arrays for the constraint coefficients.
 
+.. function:: delvars!(m::AbstractLinearQuadraticModel, idxs)
+
+    Removes the variables with indices in the vector idxs. The remaining variables in the model
+    are renumbered. Let, for example, before deletion there be six variables with indices 
+    1, 2, 3, 4, 5, 6, suppose we delete indices 1, 2, and 4, then there will be 3 variables remaining
+    in the model, those which were previously numbered 3, 5, and 6. Now these 3 have indices
```
Comment (blegat) [inline doc/lpqcqp.rst]: Wouldn't "respectly have" be even more precise ?

**Comment by joaquimg** [comment]
Anything else here?

File: doc/lpqcqp.rst [hunk 8]
```diff
@@ -98,8 +98,8 @@ to indicate equality constraints.
     Removes the variables with indices in the vector idxs. The remaining variables in the model
     are renumbered. Let, for example, before deletion there be six variables with indices 
     1, 2, 3, 4, 5, 6, suppose we delete indices 1, 2, and 4, then there will be 3 variables remaining
-    in the model, those which were previously numbered 3, 5, and 6. Now these 3 have indices
-    1, 2 and 3.
+    in the model, those which were previously numbered 3, 5, and 6. Now these 3 remaining variable
```
Comment (joaquimg) [inline doc/lpqcqp.rst]: My bad, should I create a new PR for this?

Comment (mlubin) [inline doc/lpqcqp.rst]: No problem, just pushed a fix

Comment (joaquimg) [inline doc/lpqcqp.rst]: thanks!

**Pull request merged**